app/models/link.rb

```diff
@@ -0,0 +1,10 @@
+class Link < ApplicationRecord
+  belongs_to :paper
+  validates_presence_of :url
+  validate :valid_url
```

Comment:
The reason I'm curious is because I think it translates it to a frontend regex which prevents the form from even being sent to the backend if it doesn't validate. It's a bit more efficient.

Either that, or I'm misremembering and that's only if you're using a particular gem.